moengage-segment-kotlin-destination/src/main/java/com/segment/analytics/kotlin/destinations/moengage/MoEngageDestination.kt

```diff
@@ -0,0 +1,250 @@
+import com.moengage.core.internal.USER_ATTRIBUTE_USER_MOBILE
+import com.moengage.core.internal.USER_ATTRIBUTE_USER_NAME
+import com.moengage.core.internal.integrations.MoEIntegrationHelper
+import com.moengage.core.internal.integrations.MoEIntegrationHelper.Companion.addIntegrationMeta
+import com.moengage.core.internal.logger.Logger
+import com.moengage.core.internal.model.IntegrationMeta
+import com.moengage.core.model.GeoLocation
+import com.moengage.core.model.IntegrationPartner
+import com.segment.analytics.kotlin.android.utilities.toJSONObject
+import com.segment.analytics.kotlin.core.AliasEvent
+import com.segment.analytics.kotlin.core.BaseEvent
+import com.segment.analytics.kotlin.core.IdentifyEvent
+import com.segment.analytics.kotlin.core.Settings
+import com.segment.analytics.kotlin.core.Storage
+import com.segment.analytics.kotlin.core.TrackEvent
+import com.segment.analytics.kotlin.core.Traits
+import com.segment.analytics.kotlin.core.platform.DestinationPlugin
+import com.segment.analytics.kotlin.core.platform.Plugin
+import com.segment.analytics.kotlin.core.platform.VersionedPlugin
+import com.segment.analytics.kotlin.core.utilities.getDouble
+import com.segment.analytics.kotlin.core.utilities.getString
+import com.segment.analytics.kotlin.core.utilities.mapTransform
+import com.segment.analytics.kotlin.core.utilities.toContent
+import kotlinx.serialization.json.jsonObject
+import org.json.JSONObject
+
+private const val MOENGAGE_SEGMENT_INTEGRATION_KEY = "MoEngage"
+private const val INTEGRATION_META_TYPE = "segment"
+private const val USER_TRAIT_ANONYMOUS_ID = "anonymousId"
+private const val USER_TRAIT_USER_EMAIL = "email"
+private const val USER_TRAIT_UNIQUE_ID = "userId"
+private const val USER_TRAIT_NAME = "name"
+private const val USER_TRAIT_USER_MOBILE = "phone"
+private const val USER_TRAIT_FIRST_NAME = "firstName"
+private const val USER_TRAIT_LAST_NAME = "lastName"
+private const val USER_TRAIT_USER_GENDER = "gender"
+private const val USER_TRAIT_USER_BDAY = "birthday"
+private const val USER_TRAIT_USER_ADDRESS = "address"
+private const val USER_TRAIT_USER_ADDRESS_CITY = "city"
+private const val USER_TRAIT_USER_ADDRESS_STATE = "state"
+private const val USER_TRAIT_USER_ADDRESS_COUNTRY = "country"
+private const val USER_TRAIT_USER_LOCATION = "location"
+private const val USER_TRAIT_USER_LOCATION_LONGITUDE = "longitude"
+private const val USER_TRAIT_USER_LOCATION_LATITUDE = "latitude"
+private const val USER_ATTRIBUTE_SEGMENT_ID = "USER_ATTRIBUTE_SEGMENT_ID"
+
+@kotlinx.serialization.Serializable
+data class MoEngageSettings(val apiKey: String)
+
+class MoEngageDestination(private val application: Application) : DestinationPlugin(),
+    VersionedPlugin {
+
+    private lateinit var integrationHelper: MoEIntegrationHelper
+    private lateinit var instanceId: String
+
+
+    companion object {
+        private const val tag = "MoEngageDestination_" + BuildConfig.MOENGAGE_SEGMENT_KOTLIN_VERSION
```

Comment:
This java style string concatenation, use kotlin's string interpolation instead

Reply:
Done